lib/stores/FileStore.ts

```diff
@@ -188,4 +193,48 @@ export default class FileStore extends DataStore {
 
     this.configstore.set(file_id, file)
   }
+
+  async deleteExpired(): Promise<number> {
+    const now = new Date()
+    const toDelete: Promise<void>[] = []
+    const fileStats = await Promise.all(
+      Object.keys(this.configstore.all).map((file_id) => {
+        return this.getUpload(file_id)
+      })
+    )
+
+    let index = 0
+    for (const file_id of Object.keys(this.configstore.all)) {
+      try {
+        const stats = fileStats[index]
+        const info = this.configstore.get(file_id)
+        if (
+          info &&
+          'creation_date' in info &&
+          this.getExpiration() > 0 &&
+          stats.size !== info.upload_length &&
+          stats.id === info.id
+        ) {
+          const creation = new Date(info.creation_date as Date)
+          const expires = new Date(creation.getTime() + this.getExpiration())
+          if (now > expires) {
+            toDelete.push(this.remove(file_id))
+            this.configstore.delete(file_id)
```

Comment:
I don't think it is safe to remove an upload from configstore while `remove`'s promise is still pending.  Current implementation of `remove` does not check configstore but I would not rely on this behavior.

When sucsessfully completed `remove` will remove file from configstore.